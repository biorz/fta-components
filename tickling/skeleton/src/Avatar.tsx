import * as React from 'react'
import omit from 'rc-util/lib/omit'
import { View } from '@tarojs/components'
import classNames from 'classnames'
import Element, { SkeletonElementProps } from './Element'

export interface AvatarProps extends Omit<SkeletonElementProps, 'shape'> {
  shape?: 'circle' | 'square'
}

const SkeletonAvatar = (props: AvatarProps) => {
  const { className, active } = props
  const prefixCls = 'fta-skeleton'
  const otherProps = omit(props, ['prefixCls', 'className'])
  const cls = classNames(
    prefixCls,
    `${prefixCls}-element`,
    {
      [`${prefixCls}-active`]: active,
    },
    className
  )
  return (
    <View className={cls}>
      <Element prefixCls={`${prefixCls}-avatar`} {...otherProps} />
    </View>
  )
}

SkeletonAvatar.defaultProps = {
  size: 'default',
  shape: 'circle',
}

export default SkeletonAvatar
