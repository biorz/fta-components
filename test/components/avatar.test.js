import React from 'react'
import { renderToString } from 'nerv-server'
import Avatar from '../../lib/components/avatar'

describe('Avatar Snap', () => {
  it('render Avatar -- props size(large) ', () => {
    const component = renderToString(<Avatar size='large' />)
    expect(component).toMatchSnapshot()
  })

  it('render Avatar -- props size(normal) ', () => {
    const component = renderToString(<Avatar size='normal' />)
    expect(component).toMatchSnapshot()
  })

  it('render Avatar -- props size(small) ', () => {
    const component = renderToString(<Avatar size='small' />)
    expect(component).toMatchSnapshot()
  })

  it('render Avatar -- props circle', () => {
    const component = renderToString(<Avatar circle />)
    expect(component).toMatchSnapshot()
  })

  it('render Avatar -- props image', () => {
    const component = renderToString(<Avatar image='https://jdc.jd.com/img/100' />)
    expect(component).toMatchSnapshot()
  })

  it('render Avatar -- props text', () => {
    const component = renderToString(<Avatar text='凹凸实验室' />)
    expect(component).toMatchSnapshot()
  })
})
