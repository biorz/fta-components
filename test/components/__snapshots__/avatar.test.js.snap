// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Snap render Avatar -- props circle 1`] = `undefined`;

exports[`Avatar Snap render Avatar -- props image 1`] = `undefined`;

exports[`Avatar Snap render Avatar -- props size(large)  1`] = `undefined`;

exports[`Avatar Snap render Avatar -- props size(normal)  1`] = `undefined`;

exports[`Avatar Snap render Avatar -- props size(small)  1`] = `undefined`;

exports[`Avatar Snap render Avatar -- props text 1`] = `undefined`;
